src/node_buffer.cc

```diff
@@ -310,6 +310,70 @@ void Base64Slice(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+// bytesCopied = buffer.mask(mask, target, targetOffset, sourceStart, sourceEnd)
+void Mask(const FunctionCallbackInfo<Value> &args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  uint32_t mask32 = args[0]->Uint32Value();
+  char* mask = (char*)&mask32;
+
+  Local<Object> target = args[1]->ToObject(env->isolate());
```

Comment:
Before doing this should do `CHECK(args[1]->IsObject());` and here do `args[1].As<Object>()`. Unless you don't expect `args[1]` to always be an object?